### PR TITLE
Implement Teads on Google AMP AdNetwork

### PR DESCRIPTION
**Pull request by RonanDrouglazet**
Integrating Teads on Google amp-ad component / AdNetwork

From this intent to implement issue #2021 

Thanks !

### CODE REVIEW

**Comment by RonanDrouglazet** [comment]
I signed it!

**Comment by RonanDrouglazet** [comment]
gulp test is ok on local, have you an idea ? 
`Chrome 48.0.2564 (Mac OS X 10.11.2): Executed 1464 of 1481 (skipped 17) SUCCESS (43.13 secs / 34.193 secs) [16:14:40] Finished 'test' after 1.37 min`

**Comment by RonanDrouglazet** [comment]
i've tryed `gulp dist && gulp presubmit && gulp test --compiled` like in travis, and all green on local ..
`Chrome 48.0.2564 (Mac OS X 10.11.2): Executed 1464 of 1481 (skipped 17) SUCCESS (45.019 secs / 35.9 secs)`

**Comment by RonanDrouglazet** [comment]
I signed it!

**Comment by RonanDrouglazet** [comment]
@googlebot CLA Signed on behalf of Corporation "Teads"

**Comment by cramforce** [comment]
I've restarted your build. Might have been a flake.

File: ads/teads.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  /*eslint "google-camelcase/google-camelcase": 0*/
+  global._teads_amp = {
+    allowed_data: ['pid'],
+    data: data
+  };
+
+  validateDataExists(data, global._teads_amp.allowed_data);
+  checkData(data, global._teads_amp.allowed_data);
+
+  loadScript(global, 'https://a.teads.tv/page/' + data.pid + '/tag');
```
Comment (cramforce) [inline ads/teads.js]: please use `encodeURIComponent(data.pid)`

**Comment by cramforce** [comment]
Looks great. One comment.

File: ads/teads.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+# Teads
+
+## Example
+
+```html
+<amp-ad width=300 height=220
```
Comment (mkhatib) [inline ads/teads.md]: The [PDF](http://mobile.teads.tv/amp/inReadbyTeads.pdf) mentions that teads have 16:9 ratio? Maybe use a `16:9` ratio for width and height for the example?

Comment (mkhatib) [inline ads/teads.md]: Also @cramforce tells me that you might be able to use layout=responsive here. So maybe document that?

**Comment by mkhatib** [comment]
One thing to keep in mind regarding the collapsing behavior of [teads inread](http://mobile.teads.tv/amp/inReadbyTeads.pdf) - AMP would probably not collapse the space taken by the add initially and that space would be left with a white blank space. I haven't tested this so I am not 100% sure - @cramforce might be able to confirm this.

It might be a better experience to just disable the collapsing behavior in amp-context.

**Comment by RonanDrouglazet** [comment]
Thanks @cramforce ! I've updated my commit from your advice

Hi @mkhatib, you're totally right, our video is 16/9 but with the top 'advertising' label and bottom teads credit, the full wrapper has a 4/3 ratio 

Thanks a lot for your advice on `layout="responsive"`, it's perfect for us and solves some resizing issue, everything looks great now ! 
Regarding the collapsing behavior, in most cases, we have an endscreen with `replay` and `learn more` buttons, sometimes coupled with a collapse countdown. we are considering to always show this endscreen in amp-context. If we talk about the close button, it's a user interact, can we call context.requestResize and be sure amp-ad closes itself ? If not, it's sounds like a bug for the final user, doesn't it ?

**Comment by cramforce** [comment]
@RonanDrouglazet We allow all resizes if the ad ever had focus. This should always be the case of anything inside the ad (like a close button) was clicked/tabbed.

**Comment by cramforce** [comment]
LGTM

We will, however, hold on merging until later this week.

**Comment by RonanDrouglazet** [comment]
Perfect ! Thanks for your help @cramforce !

**Comment by RonanDrouglazet** [comment]
Hi @cramforce , 

could you give us an update concerning the merge please ? Thanks a lot !

**Comment by cramforce** [comment]
Thanks for the ping. Will merge on Monday. Would you ping rebasing and pushing again?

**Comment by RonanDrouglazet** [comment]
ok i will do it on Monday, thanks @cramforce

**Comment by RonanDrouglazet** [comment]
Hi @cramforce, i've rebased teads-adnetwork-1.0.0 branch from amphtml/master ;)

**Comment by cramforce** [comment]
Thanks!

**Pull request merged**